common/src/androidTest/java/com/microsoft/identity/common/internal/platform/AsymmetricKeyFactoryTests.java

```diff
@@ -0,0 +1,129 @@
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+public class AsymmetricKeyFactoryTests {
+
+    private static final String SAMPLE_KEY_1 = "sample_key_1";
+    private static final String SAMPLE_KEY_2 = "sample_key_2";
+
+    private AsymmetricKeyFactory mKeyFactory;
+
+    @Before
+    public void setUp() {
+        final Context context = InstrumentationRegistry.getTargetContext();
+        mKeyFactory = new AndroidKeystoreAsymmetricKeyFactory(context);
+    }
+
+    @Test
+    public void testCanGenerateAsymmetricKey() throws ClientException {
+        final AsymmetricKey asymmetricKey = mKeyFactory.generateAsymmetricKey(SAMPLE_KEY_1);
+        Assert.assertNotNull(asymmetricKey);
+        Assert.assertNotNull(asymmetricKey.getThumbprint());
+    }
+
+    @Test
+    public void testCanGenerateMultipleKeys() throws ClientException {
+        final AsymmetricKey asymmetricKey1 = mKeyFactory.generateAsymmetricKey(SAMPLE_KEY_1);
+        final AsymmetricKey asymmetricKey2 = mKeyFactory.generateAsymmetricKey(SAMPLE_KEY_2);
+
+        Assert.assertNotNull(asymmetricKey1);
+        Assert.assertNotNull(asymmetricKey1.getThumbprint());
+
+        Assert.assertNotNull(asymmetricKey1);
+        Assert.assertNotNull(asymmetricKey1.getThumbprint());
+
+        // Assert they're not the same
+        Assert.assertNotEquals(
+                asymmetricKey1.getThumbprint(),
+                asymmetricKey2.getThumbprint()
+        );
+    }
+
+    @Test
+    public void testClearingOneKeyPreservesAnother() throws ClientException {
+        final AsymmetricKey asymmetricKey1 = mKeyFactory.generateAsymmetricKey(SAMPLE_KEY_1);
+        final AsymmetricKey asymmetricKey2 = mKeyFactory.generateAsymmetricKey(SAMPLE_KEY_2);
+
+        final String key1Thumbprint = asymmetricKey1.getThumbprint();
+
+        // Clear our second key
+        mKeyFactory.clearAsymmetricKey(SAMPLE_KEY_2);
+
+        // Reload our first key
+        final AsymmetricKey reloadedFirstKey = mKeyFactory.loadAsymmetricKey(SAMPLE_KEY_1);
+
+        // Reload the first key - assert that signature hasn't changed
+        Assert.assertEquals(
+                key1Thumbprint,
+                reloadedFirstKey.getThumbprint()
+        );
+
+        // Assert the alias is the same
+        Assert.assertEquals(
+                asymmetricKey1.getAlias(),
+                reloadedFirstKey.getAlias()
+        );
+    }
+
+    @Test
+    public void testCanGenerateKeyWhenLoadCalledBeforeGeneration() throws ClientException {
+        final AsymmetricKey key1 = mKeyFactory.loadAsymmetricKey(SAMPLE_KEY_1);
+        Assert.assertEquals(SAMPLE_KEY_1, key1.getAlias());
+    }
+
+    @Test
+    public void testCanClearAsymmetricKey() throws ClientException {
+        // Generate a key
+        final AsymmetricKey asymmetricKey1 = mKeyFactory.generateAsymmetricKey(SAMPLE_KEY_1);
```

Comment:
Maybe add a test that two calls to generate with the same alias result in the same key?

Reply:
ACK

Reply:
477a0592f132697fe089a128787d7bb7f12d189f